Bare metal greentea support

rajkan01 opened the pull request:
<!--
For more information on the requirements for pull requests, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html#pull-request-types).
-->
### Description (*required*)
Enabling green tea test support for bare metal and test cases update


<!--
    Required
    Add here detailed changes summary, testing results, dependencies
    Good example: https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html (Pull request template)
-->
##### Summary of change (*What the change is for and why*)
1. Added the baremetal.json under Tests directory to provide configuration for green tea test framework for bare metal.
2. Added the required components(mbed-crypto,mbedtls, filesystem etc.,) in baremetal.json  to make mbed os build success even though some of those test cases are not supported in bare-metal.
3. Removed the EMAC and USB_DEVICE from the build even though target have support but those components are not supported in bare metal
4. Whichever test cases heavily depend on RTOS are guarded with  MBED_CONF_RTOS_PRESENT flag so that the green tea framework will mark those test case as "Build Skips" category.
 - Note: Whichever component test cases use the RTOS API but still, those particular components (example: events) can run without RTOS so later we need to work on this issue to make those components test cases to run on bare-metal.
5. Tested the green tea test on below list of targets with bare metal config and added the logs in IOTCORE-1397[https://jira.arm.com/browse/IOTCORE-1397] Jira ticket
     - K64F
     - K66F
     - Nucleo F429ZI
     - Nucleo F411RE
     - Nucleo L073RZ
     - Nucleo F207ZG 
6. Command to run green tea test for bare metal for NUCLEO_F429ZI,K64F:
mbed test  -m NUCLEO_F429ZI -t GCC_ARM --app-config TESTS/configs/baremetal.json
mbed test  -m K64F -t GCC_ARM --app-config TESTS/configs/baremetal.json

##### Documentation (*Details of any document updates required*)

----------------------------------------------------------------------------------------------------------------
### Pull request type (*required*)

<!--
    Required
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front
    of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
    description heading and to add a 'x' to the correct box.
-->
    [x] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results (*required*)

<!--
    Required
    For example, add test results for new target
-->
    [] No Tests required for this change (E.g docs only update)
    [x] Covered by existing mbed-os tests (Greentea or Unittest)
    [] Tests / results supplied as part of this PR
    
    
----------------------------------------------------------------------------------------------------------------
### Reviewers (*optional*)
@evedon @hugueskamba @jamesbeyond 
<!--
    Optional
    Request additional reviewers with @username
-->

----------------------------------------------------------------------------------------------------------------
### Release Notes (*required for feature/major PRs*)

<!--
    All 3 sections are compulsory for Major PR types. For Feature PRs only the summary section is required.
    This section is automatically added to release notes. Please fill in each sub-section with sufficient detail for a user.
    For more information, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html#pull-request-types). 
-->

##### Summary of changes

##### Impact of changes

##### Migration actions required

rajkan01 committed:
Bare metal greentea support
 -Added the baremetal.json on Tests directory to enable bare-metal
 -Conditional enable of greentea metrics

Note:
Run green tea with bare metal
mbed test  -m target -t GCC_ARM -n tests-mbed_platform-transaction  --app-config TESTS/baremetal.json

0xc0170 commented on TESTS/baremetal.json:
```diff
@@ -0,0 +1,8 @@
+{ 
+    "requires": [ 
```
isn't this forcing baremetal for all tests?

rajkan01 replied:
No untill passes the command line argument --app-config with baremetal.json

evedon replied:
We should create a `TESTS/configs` folder so we can add different test configs variants.

jamesbeyond commented on features/frameworks/greentea-client/source/greentea_metrics.cpp:
```diff
@@ -20,6 +20,8 @@
 #include "platform/mbed_stats.h"
 #include <stdint.h>
 
+#if defined(MBED_CONF_RTOS_PRESENT) 
```
This `#if` define would be a bit overlap on the logical  with later `line 35`, I think you can combine those `#if`  together

evedon replied:
I think you can get CPU stats in baremetal if the device has DEVICE_SLEEP and DEVICE_LPTICKER.

rajkan01 replied:
Moved the RTOS supported metrics under MBED_CONFIG_RTOS_PRESENT macro to indirectly disable for bare metal.

evedon commented on features/frameworks/greentea-client/source/greentea_test_env.cpp:
```diff
@@ -67,7 +67,9 @@ static void greentea_write_string(const char *str);
  *           This function is blocking.
  */
 void _GREENTEA_SETUP_COMMON(const int timeout, const char *host_test_name, char *buffer, size_t size) {
+#if defined(MBED_CONF_RTOS_PRESENT)	
```
We should be able to get some metrics in baremetal, for example CPU stats. Rather than disabling the call to `greentea_metrics_setup` disable the metrics that are not available withour an RTOS in `greentea_metrics_setup`.

rajkan01 replied:
Yes Done

rajkan01 committed:
Fixed the greentea test dependency and incorporated the review comments
 -Added the mbedtls,crypto,psa,filesystem,fat,littlefs in baremetal.json to resolve compiler issue
 -Disable metrics that are not available for bare metal
 -Moved the baremetal.json inside TESTS/configs directory

rajkan01 committed:
- Emac,NFC-EEPROM,PSA-CRYPTO components build issue fixed
 - RTOS supported test case guarded with MBED_CONF_RTOS_PRESENT flag
 - Added the #error to non supported test case for bare metal

jamesbeyond reviewed (changes requested):
Discussed with @rajkan01, 155 changes file would too much for review.
Suggested to breakdown in to smaller PRs, and let individual team-leads to review

rajkan01 committed:
-New PR for PSA,Netsocket,storage components,reverted all the changes to back to original
 -Change the #error description more precisely
 -Separated the MBED_CONF_RTOS_PRESENT guarded

evedon commented on TESTS/configs/baremetal.json:
```diff
@@ -0,0 +1,40 @@
+	"kv-config",
+	"events",
+	"kv-global-api",
+	"sd"
```
Please correct teh indentation.

evedon commented on TESTS/events/timing/main.cpp:
```diff
@@ -84,6 +86,7 @@ void timer_timing_test()
 // equeue tick timing test
 void tick_timing_test()
 {
+#if defined(MBED_CONF_RTOS_PRESENT)
```
This test should work without an RTOS as there is an implementation of `equeue_tick` without an RTOS.
I think it is worth understanding why the test was failing.

evedon commented on TESTS/events/timing/main.cpp:
```diff
@@ -96,11 +99,13 @@ void tick_timing_test()
 // equeue semaphore timing test
 void semaphore_timing_test()
 {
+#if defined(MBED_CONF_RTOS_PRESENT)
```
This test should also work without an RTOS

evedon commented on TESTS/integration/COMMON/file_test.cpp:
```diff
@@ -103,3 +104,4 @@ void file_test_read(const char *file, size_t offset, const unsigned char *data,
            float(data_length) / 1024, float(data_length) / timer.read() / 1024, timer.read());
 }
 
+#endif //#if INTEGRATION_TESTS
```
Add a new line at EOF

evedon commented on TESTS/integration/COMMON/file_test.h:
```diff
@@ -20,7 +20,8 @@
 void file_test_write(const char *file, size_t offset, const unsigned char *data, size_t data_length, size_t block_size);
 
 void file_test_read(const char *file, size_t offset, const unsigned char *data, size_t data_length, size_t block_size);
+#endif
```
Add a new line at EOF

evedon commented on TESTS/mbed_drivers/crc/main.cpp:
```diff
@@ -136,6 +136,7 @@ void test_thread(void)
 
 void test_thread_safety()
 {
+#if defined(MBED_CONF_RTOS_PRESENT)
```
The entire test should be disabled otherwise it will be misaleading as it will look like that test test was run.

evedon commented on TESTS/mbed_drivers/lp_ticker/main.cpp:
```diff
@@ -71,6 +71,7 @@ void increment_multi_counter(void)
  */
 void test_multi_ticker(void)
 {
+#if defined(MBED_CONF_RTOS_PRESENT)
```
I think this test should run without an RTOS. Replace the call `ThisThread::sleep_for` by `thread_sleep_for`.

evedon commented on TESTS/mbed_drivers/lp_ticker/main.cpp:
```diff
@@ -140,6 +142,7 @@ void test_multi_call_time(void)
  */
 void test_detach(void)
 {
+#if defined(MBED_CONF_RTOS_PRESENT)
```
Disable the entire test rather than providing an empty function.

evedon commented on TESTS/mbed_drivers/lp_timer/main.cpp:
```diff
@@ -19,7 +19,9 @@
 #include "greentea-client/test_env.h"
 #include "unity.h"
 #include "utest.h"
+#if defined(MBED_CONF_RTOS_PRESENT)
```
The include is probably not required for this test if this is the only line that needs removing.

evedon commented on TESTS/mbed_drivers/rtc/main.cpp:
```diff
@@ -18,7 +18,9 @@
 #include "greentea-client/test_env.h"
 #include "unity.h"
 #include "utest.h"
+#if defined(MBED_CONF_RTOS_PRESENT)
```
The include `rtos.h` can probably be removed.

evedon commented on TESTS/mbed_drivers/ticker/main.cpp:
```diff
@@ -193,6 +194,7 @@ void test_case_2x_ticker()
  */
 void test_multi_ticker(void)
 {
+#if defined(MBED_CONF_RTOS_PRESENT)
```
This test should be enabled. Use `thread_sleep_for`

evedon commented on TESTS/mbed_drivers/ticker/main.cpp:
```diff
@@ -262,6 +265,7 @@ void test_multi_call_time(void)
  */
 void test_detach(void)
 {
+#if defined(MBED_CONF_RTOS_PRESENT)
```
Disable the entire test.

evedon commented on TESTS/mbed_drivers/timeout/timeout_tests.h:
```diff
@@ -15,6 +15,7 @@
  */
 #ifndef MBED_TIMEOUT_TESTS_H
 #define MBED_TIMEOUT_TESTS_H
+#if defined(MBED_CONF_RTOS_PRESENT)
```
I don't think this is required because you added:
```
#if !defined(MBED_CONF_RTOS_PRESENT)
#error [NOT_SUPPORTED] usticker test requires RTOS to run.
#else
```
in main.cpp at the top of the file.

evedon commented on TESTS/mbed_drivers/timer/main.cpp:
```diff
@@ -19,7 +19,9 @@
 #include "greentea-client/test_env.h"
 #include "unity.h"
 #include "utest.h"
+#if defined(MBED_CONF_RTOS_PRESENT)
```
You should be able to remove the include.

evedon commented on TESTS/mbed_hal/rtc/main.cpp:
```diff
@@ -120,6 +120,7 @@ void rtc_sleep_test()
 /* Test that the RTC keeps counting even after ::rtc_free has been called. */
 void rtc_persist_test()
 {
+#if defined(MBED_CONF_RTOS_PRESENT)
```
This test should be enabled. Replace the call `ThisThread::sleep_for` by `thread_sleep_for`.

evedon commented on TESTS/mbed_hal/rtc/main.cpp:
```diff
@@ -156,6 +158,7 @@ void rtc_glitch_test()
 /* Test that the RTC correctly handles different time values. */
 void rtc_range_test()
 {
+#if defined(MBED_CONF_RTOS_PRESENT)
```
Same comment: this test should be enabled.

evedon commented on TESTS/mbed_hal/sleep/main.cpp:
```diff
@@ -14,6 +14,10 @@
  */
 
+#if !defined(MBED_CONF_RTOS_PRESENT)
+#error [NOT_SUPPORTED] sleep test cases requires RTOS to run.
```
We should try to get this test enabled.

evedon commented on TESTS/mbed_hal/sleep_manager/main.cpp:
```diff
@@ -13,6 +13,10 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+#if !defined(MBED_CONF_RTOS_PRESENT)
```
Same here. I'd like to get this test enabled.

evedon commented on TESTS/mbed_hal/sleep_manager_racecondition/main.cpp:
```diff
@@ -26,7 +26,7 @@
 
 #define TEST_STACK_SIZE 256
-
+#if defined(MBED_CONF_RTOS_PRESENT)
```
Replace `ThisThread::sleep_for` and remove the flag

evedon commented on TESTS/mbed_hal/sleep_manager_racecondition/main.cpp:
```diff
@@ -35,9 +35,10 @@ void sleep_manager_locking_thread_test()
+#endif
 void sleep_manager_multithread_test()
 {
+#if defined(MBED_CONF_RTOS_PRESENT)
```
Disable the entire function

evedon commented on TESTS/mbed_hal/watchdog/main.cpp:
```diff
@@ -14,6 +14,10 @@
  * limitations under the License.
  */
+#if !defined(MBED_CONF_RTOS_PRESENT)
+#error [NOT_SUPPORTED] Watchdog test cases requires RTOS to run.
```
I'd like this test enabled. What RTOS features are required?

rajkan01 replied:
Watchdog uses thread to reset hardware counter periodically, This needs further test cases understanding and investigation.I have raised ticket IOTCORE-1438 to fix this.

evedon commented on TESTS/mbed_hal/watchdog_reset/main.cpp:
```diff
@@ -14,6 +14,10 @@
  * limitations under the License.
  */
+#if !defined(MBED_CONF_RTOS_PRESENT)
+#error [NOT_SUPPORTED] Watchdog reset test cases requires RTOS to run.
```
I'd like this test enabled. What RTOS features are required?

evedon commented on TESTS/mbed_platform/crash_reporting/main.cpp:
```diff
@@ -13,6 +13,9 @@
  * limitations under the License.
  */
+#if !defined(MBED_CONF_RTOS_PRESENT)
+#error [NOT_SUPPORTED] crash_reporting test cases requires RTOS to run.
```
I'd like this test enabled. What RTOS features are required?

rajkan01 replied:
After the fatal error injected from green tea test than on the next target boot mbed error component suppose to have information on last fatal error details but it seems to last fatal error information lost and the callback to mbed_error_reboot_callback is not happening. I have raised the ticket  IOTCORE-1439 to fix this.

evedon commented on TESTS/mbed_platform/error_handling/main.cpp:
```diff
@@ -118,6 +118,7 @@ void test_error_capturing()
  */
 void test_error_context_capture()
 {
+#if defined(MBED_CONF_RTOS_PRESENT)
```
Only disable the code that uses a thread.

evedon commented on TESTS/mbed_platform/error_handling/main.cpp:
```diff
@@ -246,16 +248,19 @@ void test_error_logging_multithread()
  */
 void test_error_hook()
 {
+#if defined(MBED_CONF_RTOS_PRESENT)
```
Disable the entire test.

evedon commented on TESTS/mbedmicro-rtos-mbed/heap_and_stack/main.cpp:
```diff
@@ -15,8 +15,8 @@
-#if defined(TARGET_CORTEX_A) || !DEVICE_USTICKER
-#error [NOT_SUPPORTED] This function not supported for this target
+#if defined(TARGET_CORTEX_A) || !DEVICE_USTICKER || !defined(MBED_CONF_RTOS_PRESENT)
+#error [NOT_SUPPORTED] test not supported.
```
It would be good to make this test run for bare metal as well. A separate task has been reaised for this work so we will work on it in a different PR.

evedon commented on TESTS/mbedmicro-rtos-mbed/malloc/main.cpp:
```diff
@@ -218,4 +221,4 @@ int main()
 }
 
-#endif // defined(MBED_RTOS_SINGLE_THREAD) || !DEVICE_USTICKER
+#endif // defined(MBED_RTOS_SINGLE_THREAD) || !DEVICE_USTICKER || !defined(MBED_CONF_RTOS_PRESENT)
```
Remove | !defined(MBED_CONF_RTOS_PRESENT)

evedon commented on TESTS/mbedmicro-rtos-mbed/malloc/main.cpp:
```diff
@@ -68,6 +69,7 @@ void task_using_malloc(void)
  */
 void test_multithread_allocation(void)
 {
+#if defined(MBED_CONF_RTOS_PRESENT)
```
Disable the entire test

evedon commented on TESTS/mbedmicro-rtos-mbed/malloc/main.cpp:
```diff
@@ -19,14 +19,15 @@
 
 extern uint32_t mbed_heap_size;
 static const int test_timeout = 25;
+#if defined(MBED_CONF_RTOS_PRESENT)
```
You should be able to keep all this code even if there is no RTOS.

evedon commented on TESTS/mbedmicro-rtos-mbed/threads/LockGuard.h:
```diff
@@ -17,6 +17,7 @@
 
 #ifndef MBEDMICRO_RTOS_MBED_THREADS_LOCK_GUARD
 #define MBEDMICRO_RTOS_MBED_THREADS_LOCK_GUARD
+#if defined(MBED_CONF_RTOS_PRESENT)
```
I don't think that this is needed as you are already checking the RTOS present flag in TESTS/mbedmicro-rtos-mbed/threads/main.cpp

evedon commented on TESTS/mbedmicro-rtos-mbed/threads/SynchronizedIntegral.h:
```diff
@@ -17,6 +17,7 @@
 
 #ifndef MBEDMICRO_RTOS_MBED_THREADS_SYNCHRONIZED_INTEGRAL
 #define MBEDMICRO_RTOS_MBED_THREADS_SYNCHRONIZED_INTEGRAL
+#if defined(MBED_CONF_RTOS_PRESENT)
```
I don't think that this is needed as you are already checking the RTOS present flag in TESTS/mbedmicro-rtos-mbed/threads/main.cpp

evedon commented on TESTS/usb_device/basic/USBEndpointTester.cpp:
```diff
@@ -169,8 +171,11 @@ USBEndpointTester::USBEndpointTester(USBPhy *phy, uint16_t vendor_id, uint16_t p
     }
     MBED_ASSERT(resolver.valid());
-
+#if defined(MBED_CONF_RTOS_PRESENT)
```
Not required since you already checked at the beginning of the file.

evedon commented on TESTS/usb_device/basic/USBEndpointTester.cpp:
```diff
@@ -861,3 +866,4 @@ void USBEndpointTester::start_ep_in_abort_test()
     write_start(_endpoints[EP_BULK_IN], _endpoint_buffs[EP_BULK_IN], (*_endpoint_configs)[EP_BULK_IN].max_packet);
     write_start(_endpoints[EP_INT_IN], _endpoint_buffs[EP_INT_IN], (*_endpoint_configs)[EP_INT_IN].max_packet);
 }
+#endif
```
Add a new line

evedon commented on TESTS/usb_device/basic/USBEndpointTester.h:
```diff
@@ -17,6 +17,7 @@
 
 #ifndef USB_ENDPOINT_TESTER_H
 #define USB_ENDPOINT_TESTER_H
+#if defined(MBED_CONF_RTOS_PRESENT)
```
Are you sure it is necessary to compile this out? I think it might or should not be included by the source files of there is no RTOS.

evedon commented on TESTS/usb_device/basic/USBTester.cpp:
```diff
@@ -56,8 +58,11 @@ USBTester::USBTester(USBPhy *phy, uint16_t vendor_id, uint16_t product_id, uint1
     int_in = resolver.endpoint_in(USB_EP_TYPE_INT, 64);
     int_out = resolver.endpoint_out(USB_EP_TYPE_INT, 64);
     MBED_ASSERT(resolver.valid());
+#if defined(MBED_CONF_RTOS_PRESENT)
```
You have already compiled this file out if there is no RTOS.

evedon commented on TESTS/usb_device/basic/USBTester.h:
```diff
@@ -17,6 +17,7 @@
 
 #ifndef USB_TESTER_H
 #define USB_TESTER_H
+#if defined(MBED_CONF_RTOS_PRESENT)
```
Are you sure it is necessary to compile this out? I think it might or should not be included by the source files of there is no RTOS.

evedon commented on TESTS/usb_device/msd/TestUSBMSD.h:
```diff
@@ -17,6 +17,8 @@
 #ifndef Test_USBMSD_H
 #define Test_USBMSD_H
 
+#if defined(MBED_CONF_RTOS_PRESENT)
```
I don't think this file is included by any source file if there is no RTOS.

evedon reviewed (changes requested):
I think we have to enable more bare metal tests. If it does not take too much time we will do it in this PR; if not we will raise tickets to work on it separately.

VeijoPesonen commented on TESTS/configs/baremetal.json:
```diff
@@ -0,0 +1,40 @@
+{ 
```
Add "rtos-api" here and there is no need to drop sleep calls in PR #11825. That's at least how it's done in [Mbed Bootloader](https://github.com/ARMmbed/mbed-bootloader/blob/master/mbed_app.json)

evedon replied:
Yes! That will help enable many more tests.

rajkan01 committed:
Incorporated the review comments
-Added "rtos-api" in TESTS/configs/baremetal.json
-some test case used thread_sleep_for instead Thisthread::sleep_for
-Added the EOF character
-Disable completely the non supported test cases

rajkan01 committed:
Enabled the sleep,sleep_manager,watchdog_reset test case for bare metal greentea test

evedon commented on TESTS/integration/COMMON/download_test.h:
```diff
@@ -20,6 +20,7 @@
  * Based on mbed-stress-test by Marcus Chang @ Arm Mbed - http://github.com/ARMmbed/mbed-stress-test
 */
-
+#if INTEGRATION_TESTS
```
You should not need to add the flag here. 
Make sure that the file is not in included by any source file. You might need to move
```
#if !INTEGRATION_TESTS
#error [NOT_SUPPORTED] integration tests not enabled for this target
#endif
```
at the top of some main.cpp

evedon commented on TESTS/integration/COMMON/file_test.h:
```diff
@@ -20,7 +20,8 @@
  * Based on mbed-stress-test by Marcus Chang @ Arm Mbed - http://github.com/ARMmbed/mbed-stress-test
 */
-
+#if INTEGRATION_TESTS
```
Same comment.

You should not need to add the flag here.
Make sure that the file is not in included by any source file. You might need to move

#if !INTEGRATION_TESTS
#error [NOT_SUPPORTED] integration tests not enabled for this target
#endif
at the top of some main.cpp

evedon commented on TESTS/integration/fs-single/main.cpp:
```diff
@@ -162,3 +162,4 @@ int main()
 
     return !Harness::run(specification);
 }
+#endif
```
Add 
```
#endif // INTEGRATION_TESTS
```
This applies to other files as well.

evedon commented on TESTS/mbed_drivers/race_test/main.cpp:
```diff
@@ -14,6 +14,11 @@
  */
+
+#if defined(MBED_RTOS_SINGLE_THREAD) || !DEVICE_USTICKER || !defined(MBED_CONF_RTOS_PRESENT)
+#error [NOT_SUPPORTED] Test not supported for single threaded and also bare metal enviroment. UsTicker need to be enabled for this test.
```
Typo: enviroment --> environment, UsTicker need --> `Us ticker needs`
But instead of `Test not supported for single threaded and also bare metal enviroment.` I would say `Race test cases require a RTOS.`

rajkan01 replied:
I already pushed the changes of splitting this into 2 as like 
`#if defined(MBED_RTOS_SINGLE_THREAD) || !defined(MBED_CONF_RTOS_PRESENT)
#error [NOT_SUPPORTED] Race test test cases require RTOS with multithread to run
#else

#if !DEVICE_USTICKER
#error [NOT_SUPPORTED] UsTicker need to be enabled for this test.
#else`

jamesbeyond commented on TESTS/usb_device/msd/main.cpp:
```diff
@@ -14,6 +14,11 @@
  */
+
+#if !defined(MBED_CONF_RTOS_PRESENT)
+#error [NOT_SUPPORTED] USB test cases are not supported on Bare metal
```
please change the error message something like: to `USB stack and tests require RTOS to run`
and please add same macro guard all usb tests: `HID` and `serial` etc.

jamesbeyond commented on TESTS/mbed_drivers/race_test/main.cpp:
```diff
@@ -14,6 +14,11 @@
+
+#if defined(MBED_RTOS_SINGLE_THREAD) || !DEVICE_USTICKER || !defined(MBED_CONF_RTOS_PRESENT)
+#error [NOT_SUPPORTED] Test not supported for single threaded and also bare metal enviroment. UsTicker need to be enabled for this test.
+#else
```
Would be better to breakdown the qualifier macro checks like:
```cpp
#if defined(MBED_RTOS_SINGLE_THREAD) || !defined(MBED_CONF_RTOS_PRESENT)
#error [NOT_SUPPORTED] Test requires RTOS and threads to run
#elif !DEVICE_USTICKER 
#error [NOT_SUPPORTED] USTicker is need to be enabled to run this test
#else
```

jamesbeyond commented on TESTS/mbedmicro-rtos-mbed/basic/main.cpp:
```diff
@@ -19,7 +19,7 @@
 
-#if defined(SKIP_TIME_DRIFT_TESTS) || defined(MBED_RTOS_SINGLE_THREAD) || !DEVICE_USTICKER
+#if defined(SKIP_TIME_DRIFT_TESTS) || defined(MBED_RTOS_SINGLE_THREAD) || !DEVICE_USTICKER || !defined(MBED_CONF_RTOS_PRESENT)
 #error [NOT_SUPPORTED] test not supported
```
please breakdown the qualifier macro checks:
```cpp
#if defined(SKIP_TIME_DRIFT_TESTS) || 
#error [NOT_SUPPORTED] Time drifting tests skipped
#elif defined(MBED_RTOS_SINGLE_THREAD) || !defined(MBED_CONF_RTOS_PRESENT)
#error [NOT_SUPPORTED] Test requires RTOS and threads to run
#elif !DEVICE_USTICKER
#error [NOT_SUPPORTED] USTicker is need to be enabled to run this test
```

rajkan01 replied:
Discussed with Jamesbeyond and split into two ifdef

jamesbeyond commented on TESTS/mbedmicro-rtos-mbed/event_flags/main.cpp:
```diff
@@ -22,7 +22,7 @@
 using utest::v1::Case;
 
-#if defined(MBED_RTOS_SINGLE_THREAD) || !DEVICE_USTICKER
+#if defined(MBED_RTOS_SINGLE_THREAD) || !DEVICE_USTICKER || !defined(MBED_CONF_RTOS_PRESENT)
```
please breakdown the qualifier macro checks like above comments

jamesbeyond commented on TESTS/mbedmicro-rtos-mbed/mail/main.cpp:
```diff
@@ -13,16 +13,17 @@
+
+#if defined(MBED_RTOS_SINGLE_THREAD) || !DEVICE_USTICKER || !defined(MBED_CONF_RTOS_PRESENT)
+#error [NOT_SUPPORTED] test not supported
+#else
```
please breakdown the qualifier macro checks like above comments

jamesbeyond commented on TESTS/mbedmicro-rtos-mbed/mutex/main.cpp:
```diff
@@ -13,16 +13,17 @@
+
+#if defined(MBED_RTOS_SINGLE_THREAD) || !DEVICE_USTICKER || !defined(MBED_CONF_RTOS_PRESENT)
+#error [NOT_SUPPORTED] test not supported
+#else
```
please breakdown the qualifier macro checks like above comments

jamesbeyond commented on TESTS/mbedmicro-rtos-mbed/semaphore/main.cpp:
```diff
@@ -13,6 +13,11 @@
+
+#if defined(MBED_RTOS_SINGLE_THREAD) || !DEVICE_USTICKER || !defined(MBED_CONF_RTOS_PRESENT)
+#error [NOT_SUPPORTED] test not supported
+#else
```
please breakdown the qualifier macro checks like above comments

jamesbeyond commented on TESTS/mbedmicro-rtos-mbed/signals/main.cpp:
```diff
@@ -20,7 +20,7 @@
-#if defined(MBED_RTOS_SINGLE_THREAD) || !DEVICE_USTICKER
+#if defined(MBED_RTOS_SINGLE_THREAD) || !DEVICE_USTICKER || !defined(MBED_CONF_RTOS_PRESENT)
 #error [NOT_SUPPORTED] test not supported
 #else
```
please breakdown the qualifier macro checks like above comments

jamesbeyond commented on TESTS/mbedmicro-rtos-mbed/threads/main.cpp:
```diff
@@ -13,6 +13,11 @@
+
+#if defined(MBED_RTOS_SINGLE_THREAD) || !DEVICE_USTICKER || !defined(MBED_CONF_RTOS_PRESENT)
+#error [NOT_SUPPORTED] test not supported
+#else
```
please breakdown the qualifier macro checks like above comments

rajkan01 committed:
Incorporated review comments

jamesbeyond reviewed (approved):
LGTM now

evedon commented on TESTS/mbedmicro-rtos-mbed/systimer/main.cpp:
```diff
@@ -19,6 +19,9 @@
 #include "utest.h"
 #include "ticker_api.h"
+#if !defined(MBED_CONF_RTOS_PRESENT)
+#error [NOT_SUPPORTED] systimer test cases requires RTOS to run.
```
Check if this test can runn now that you have rtos-api

rajkan01 committed:
Error message correction and rx_lib header filer removed on systimer test case

evedon commented on TESTS/mbedmicro-rtos-mbed/systimer/main.cpp:
```diff
@@ -20,9 +20,12 @@
 extern "C" {
 #include "rtx_lib.h"
 }
+#endif
```
You can probably remove if the rest of the file is fine.

VeijoPesonen reviewed (approved):
Looks good to me

0xc0170 reviewed (approved):
As stated, we will need to fix some tests that should run with baremetal (everything in platform/hal should do). This comes as result "rtos always included".
